receiver/receiver.go

```diff
@@ -60,23 +65,23 @@ type Factory interface {
 
 	// CreateTracesReceiver creates a TracesReceiver based on this config.
 	// If the receiver type does not support tracing or if the config is not valid
-	// an error will be returned instead.
+	// an error will be returned instead. `nextConsumer` is never nil.
 	CreateTracesReceiver(ctx context.Context, set CreateSettings, cfg component.Config, nextConsumer consumer.Traces) (Traces, error)
 
 	// TracesReceiverStability gets the stability level of the TracesReceiver.
 	TracesReceiverStability() component.StabilityLevel
 
 	// CreateMetricsReceiver creates a MetricsReceiver based on this config.
 	// If the receiver type does not support metrics or if the config is not valid
-	// an error will be returned instead.
+	// an error will be returned instead. `nextConsumer` is never nil.
 	CreateMetricsReceiver(ctx context.Context, set CreateSettings, cfg component.Config, nextConsumer consumer.Metrics) (Metrics, error)
 
 	// MetricsReceiverStability gets the stability level of the MetricsReceiver.
 	MetricsReceiverStability() component.StabilityLevel
 
 	// CreateLogsReceiver creates a LogsReceiver based on this config.
 	// If the receiver type does not support the data type or if the config is not valid
-	// an error will be returned instead.
+	// an error will be returned instead. `nextConsumer` is never nil.
```

Comment:
not sure i understand this comment, are you saying that `nextConsumer` must never be nil? The comment doesn't make that clear

Reply:
In the past implementors would check that nextConsumer may be nil. We are making the strong guarantee to implementors that `nextConsumer` is never nil and there's no need to check and return errors for it.